presto-main/src/main/java/com/facebook/presto/execution/QueryStats.java

```diff
@@ -373,12 +373,18 @@ public static QueryStats create(
                 PlanFragment plan = stageInfo.getPlan().get();
                 for (OperatorStats operatorStats : stageExecutionStats.getOperatorSummaries()) {
                     // NOTE: we need to literally check each operator type to tell if the source is from table input or shuffled input. A stage can have input from both types of source.
+                    // NOTE: also check for operator names from presto native execution.
                     String operatorType = operatorStats.getOperatorType();
-                    if (operatorType.equals(ExchangeOperator.class.getSimpleName()) || operatorType.equals(MergeOperator.class.getSimpleName())) {
+                    if (operatorType.equals(ExchangeOperator.class.getSimpleName()) ||
+                            operatorType.equals(MergeOperator.class.getSimpleName()) ||
+                            operatorType.equals("Exchange") ||
+                            operatorType.equals("MergeExchange")) {
                         shuffledPositions += operatorStats.getRawInputPositions();
                         shuffledDataSize += operatorStats.getRawInputDataSize().toBytes();
                     }
-                    else if (operatorType.equals(TableScanOperator.class.getSimpleName()) || operatorType.equals(ScanFilterAndProjectOperator.class.getSimpleName())) {
+                    else if (operatorType.equals(TableScanOperator.class.getSimpleName()) ||
```

Comment:
Do we want to add the native execution check here for the case without Operator suffix? Thanks!

Reply:
If we do, then for a while Native won't be checked properly here, because for now the names are being changed to Java names.
IMHO, this is fine as it is unlikely that any operator would have these names.
We can wait for other reviewers to chip in.